api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -117,37 +77,97 @@ public Condition getCondition(Integer conditionId) {
 	}
 
 	/**
-     * @see ConditionService#getAllConditions(Patient)
+	 * @see ConditionService#getAllConditions(Patient)
 	 */
 	@Override
 	@Transactional(readOnly = true)
 	public List<Condition> getAllConditions(Patient patient) {
 		return conditionDAO.getAllConditions(patient);
 	}
 
+	/**
+	 * @see ConditionService#getConditionsByEncounter(Encounter)
+	 */
+	@Override
+	public List<Condition> getConditionsByEncounter(Encounter encounter) throws APIException {
+		return conditionDAO.getConditionsByEncounter(encounter);
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 * @return the saved condition
+	 */
+	@Override
+	public Condition saveCondition(Condition condition) {
+		Condition existingCondition = Context.getConditionService().getConditionByUuid(condition.getUuid());
+		// If there is no existing condition, then we are creating a condition
+		if (existingCondition == null) {
+			return conditionDAO.saveCondition(condition);
+		}
+
+		// If the incoming condition has been voided, we simply void the existing condition
+		// All other changes are ignored
+		if (condition.getVoided()) {
+			if (!existingCondition.getVoided()) {
+				return Context.getConditionService().voidCondition(existingCondition,
+					StringUtils.isNotBlank(condition.getVoidReason()) ? condition.getVoidReason() : "Condition deleted");
+			} else {
+				return existingCondition;
+			}
+		}
+		
+		// If the existing condition is voided, we will only calls to unvoid the condition
+		// All other changes are ignored
+		if (existingCondition.getVoided()) {
+			if (!condition.getVoided()) {
+				return Context.getConditionService().unvoidCondition(existingCondition);
+			} else {
+				return existingCondition;
+			}
+		}
+
+		// If we got here, the updated condition and the existing condition are both live, so the updated condition is now
+		// replacing the existing condition
+		Condition newCondition = Condition.newInstance(condition);
+		newCondition.setPreviousVersion(existingCondition);
+
+		if (!existingCondition.getVoided()) {
+			existingCondition.setVoided(true);
+			existingCondition.setVoidedBy(Context.getAuthenticatedUser());
+			existingCondition.setVoidReason("Condition replaced");
+			conditionDAO.saveCondition(existingCondition);
```

Comment:
private void voidExistingCondition(Condition existingCondition) {
		existingCondition.setVoided(true);
		existingCondition.setVoidedBy(Context.getAuthenticatedUser());
		existingCondition.setVoidReason("Condition replaced");
		conditionDAO.saveCondition(existingCondition);
	}

Reply:
I'm not sure I see the value of adding a method like that? Can you elaborate? To be clear, the reason that this logic is here is to void and replace an existing condition when updating the condition, hence the need to support voiding like that seems limited to this one method (other consumers of this service should continue to use `Context.getConditionService().voidCondition(myCondition, "My voiding reason");`.